### ISSUE TITLE
How to get rid of the space in front of a letter at the beginning of a paragraph?

### ISSUE THREAD

**Issue by muuvmuuv** [rsms/inter#216]
**Describe the bug**
I can't really tell if this is a bug or the general expected behaviour in Typography but I have this in Figma, Sketch and on Websites with Inter and also with other fonts.

When aligning a text block (12px) and a headline (42px) to the left. The headlines first letter has significant more space to the outlining box than the text block.

If this is the expected behaviour, what is the most common way to handle this inconsistency? Simply -margin the headline to the left or is there a `font-feature-setting` I need to enable?

![Bildschirmfoto 2019-11-07 um 13 32 18](https://user-images.githubusercontent.com/11534760/68395622-5f984100-0170-11ea-808e-45d5894d2f2c.png)

**To Reproduce**
1. e.g. open Figma
2. Create two text boxes with, one with 12px and one with 42px
3. align them to the left
4. Attach a ruler to the left border

You can also have a look at this Figma file from Figma itself. When you zoom in and use the macOS screenshot feature (or any other ruler tool), you can see that all text is not perfectly aligned to the left but the boxes are.

https://www.figma.com/file/akQyvw1tYQmskou0A3FKfcGk

**Expected behavior**
In my expectation it should begin at the outlining border and also end there. So aligning text to a site is even to other text blocks.

**Screenshots**

**Environment**
- OS: [e.g. macOS 10.14] macOS 10.15.1 
- App that renders the font [if applicable; e.g. Chrome, Adobe Illustrator]: Figma, Sketch, Brave browser (I guess everywhere)
- Version of font: Inter Regular. Version 3.011;git-f93a4a705

**Additional context**

Would love to read common pattern to handle these and how other designers solve this.

**Comment by rsms** [comment]
This is unfortunately how it is with fonts and typography in computers, for good reasons. What is commonly done is to adjust single lines by some offset. In graphic design, this is done on a per line basis to match up optically, which is context dependent.

There’s a poorly-supported OpenType feature called [Optical Bounds (`opbd`)](https://docs.microsoft.com/en-us/typography/opentype/spec/features_ko#opbd) which is meant to address this. But, it’s not well supported.

If you think Inter should implement `opbd` I would appreciate it if you could open a separate issue for that and include some research around what software supports it and what the limitations are :–)

Thank you

**Comment by rsms** [comment]
There are many articles and even books available on the subject.
This article has a few good examples: https://rafaltomal.com/optically-perfect/

Here’s a recreation from a part of that article, but using Inter:
<img width="801" alt="sdf9j2" src="https://user-images.githubusercontent.com/47046/69769433-ace94c00-1139-11ea-991d-993cb06ab564.png">

Depending on the arrangement, the “W” wants to be offset by different amounts.

What Inter and every other good font does, is to provide a good middle ground; a good common-case alignment. Here are the same samples but typed out as two lines, using Inter’s build-in alignment:

<img width="801" alt="dsf92" src="https://user-images.githubusercontent.com/47046/69769493-e15d0800-1139-11ea-9c8b-124a5898595c.png">

As you may notice, the alignment provided “out of the box” is pretty much a blend between the two “ideal” alignments depicted in the first illustration.

Hope this sheds some light on the challenge!

**Comment by muuvmuuv** [comment]
Thank you for the detailed explanation!! It is sad that `opbd` ist not widely supported, looks promising. 
I think something like that would help visually allign typography better in the digital world without nifty hacks. The Inter way to deal with it is nice, I guess my eyes will be the only once who will see the pixel diff/indention. But I totally agree that aligning the letters all to the same edge (also as mentioned in the article you suggeted) cause text to look not that perfectly aligned while I think that a feature that shifts them to a virtual border to look like they are perfectly aligned would make sense. I guess `opbd` is likely that feature as I understood from the MS article, but as I'm not a typo human I can't tell how much effort this would be. 

I'll do a little research about it and if I think this is something to consider I'll open a new issue with more information if needed. We use the font in our website, mobile apps and print designs. Aligning the text visually perfect is easy in print but needs hacks in mobile and web, so in the case of mobile it would be nice to have a visually perfect alligned text without hacks.

Open Source is great, thank you for creating Inter and making it the most valuable font I know for free!
